### ISSUE TITLE
Metric `pessoas_vacinadas_completamente` decreases

### ISSUE THREAD

**Issue by lucasrodes** [dssg-pt/covid19pt-data#929]
Hi,
I have realized that the metric `pessoas_vacinadas_completamente` from [vacinas.csv](https://github.com/dssg-pt/covid19pt-data/blob/master/vacinas.csv) has a decreasing value from 28-03-2021 to 29-03-2021. This is also visible with the field `pessoas_vacinadas_parcialmente_novas`, where the value for 29-03-2021 is negative.

|data      |doses  |doses_novas|doses1 |doses1_novas|doses2|doses2_novas|pessoas_vacinadas_completamente|pessoas_vacinadas_completamente_novas|pessoas_vacinadas_parcialmente|pessoas_vacinadas_parcialmente_novas|pessoas_inoculadas|pessoas_inoculadas_novas|vacinas|vacinas_novas|
|----------|-------|-----------|-------|------------|------|------------|-------------------------------|-------------------------------------|------------------------------|------------------------------------|------------------|------------------------|-------|-------------|
|28-03-2021|1599599|63415      |1130258|61897       |469341|1518        |501592                         |1518                                 |710407                        |60379                               |1211999           |61897                   |1681381|63415        |
|29-03-2021|1618399|18800      |1148757|18499       |469642|301         |500926                         |-666                                 |739635                        |29228                               |1240561           |28562                   |1710246|28865        |

Thanks!

---

Ignore the issue label

**Comment by davipt** [comment]
It is expected to see some unexpected variations on the values for the calculated pessoas_vacinadas_* and vacinas because of adjustments on the weekly data e.g. because of the islands at those weeks, with madeira and half of açores being published a week or two before, and açores being sorted out I think a week later. Also, weekly historic values are updated retroactively (with better values) so it may happen that next week this gets sorted out.

For completeness, what is happening here is that on the 22nd there are daily and weekly numbers. The difference between nacional and continente (the islands) is calculated. As well as the difference between doses1/doses2 and the respective daily/weekly values (to correct the unidoses/Janssen, which is not the case yet on this week). This adjustments carries on for the whole week, from 22 to 28. On 29 it gets adjusted again with the weekly data published that day. On paper it should always adjust for the better, as it adjusts for the latest islands values that are unknown for those six days, but in this case it adjusts down for some reason I can't pinpoint exactly why.

In other words, these values seems to be related to the date people are introduced in the system, and not really the date the vaccine occurs, so over time the numbers are correct, but looking at daily values there may be inconsistencies - days with more people than reality, or this adjustment where people were counted on prior days and then corrected on the 29th.

Does this make sense?

**Comment by davipt** [comment]
Furthermore, the week of March 22 to 29 is the week where 2nd doses of Pfizer were not given as it matches the period where the rules switched from 21 days / 3 weeks to 28 days / 4 weeks.

Considering daily values are locked in time (and known to not be exact) whilst weekly values are updated retroactively, in between dose1/dose2 adjustments (albeit Janssen was not significant at the time) and islands adjustments, as the 2nd doses were such a small value (about 20k), the weekly data of 29 readjusted them such that it became negative to correct the error on the previous six days.
